src/Components/Endpoints/src/FormMapping/Converters/FileConverter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Endpoints.FormMapping;
+
+internal sealed class FileConverter<T>(HttpContext? httpContext) : FormDataConverter<T>, ISingleValueConverter
+{
+    [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
+    internal override bool TryRead(ref FormDataReader reader, Type type, FormDataMapperOptions options, out T? result, out bool found)
+    {
+        if (httpContext == null)
+        {
+            result = default;
+            found = false;
+            return true;
+        }
+
+        if (typeof(T) == typeof(IFormFileCollection))
+        {
+            result = (T)httpContext.Request.Form.Files;
```

Comment:
Is this going to read form sync over async?

Reply:
This converter gets called during the binding phase for a minimal API after the form has been read in [TryReadFormAsync](https://source.dot.net/#Microsoft.AspNetCore.Http.Extensions/RequestDelegateFactory.cs,8ae152d4c5cb66f6) and cached so we can avoid the blocking call here.

Reply:
Add that as a comment

Reply:
How about adding:

```cs
Debug.Assert(httpContext.Features.Get<IFormFeature>().Form != null, "Form should have been loaded during binding.");
```

That double-checks the form was previously loaded.